### PR TITLE
[DOPS-1183] fix EnvoyFilter

### PR DESCRIPTION
**Pull request by volkorny**

### CODE REVIEW

File: stable/application-mesh/templates/envoy-filter.yaml [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
-{{- if $ingress.settings }}
-{{- if $ingress.settings.max_request_bytes }}
-{{- $host := required "Ingress host must be specified" $ingress.host }}
+{{- if .Values.max_request_bytes }}
 ---
 apiVersion: networking.istio.io/v1alpha3
 kind: EnvoyFilter
 metadata:
-  name: {{ include "application-helpers.name" $ }}-{{ include "application-helpers.uniq-id" $host }}
+  name: {{ include "application-helpers.name" $ }}
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/templates/envoy-filter.yaml]: I would suggest naming the filter as `Product-Feature`

File: stable/application-mesh/templates/envoy-filter.yaml [hunk 2]
```diff
@@ -1,36 +1,24 @@
...
 spec:
   workloadSelector:
     labels:
-      istio: {{ if eq (index (splitList "-" $ingress.gateway) 0) "external" }}ingress-public-gateway{{ else }}ingress-private-gateway{{ end }}
+      app.kubernetes.io/name: {{ .Values.global.application.product }}
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/templates/envoy-filter.yaml]: lets target pods that work as ingress destinations

File: stable/application-mesh/templates/envoy-filter.yaml [hunk 3]
```diff
@@ -1,36 +1,24 @@
 {{- if .Values.global.mesh.enabled }}
-{{- range $_, $ingress := .Values.ingress }}
-{{- if ne $ingress.host "example.matic.com" }}
-{{- if $ingress.settings }}
-{{- if $ingress.settings.max_request_bytes }}
-{{- $host := required "Ingress host must be specified" $ingress.host }}
+{{- if .Values.max_request_bytes }}
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/templates/envoy-filter.yaml]: Can we add a bit of structure to component settings.

**Review by volodymyr-mykhailyk — COMMENTED** [review]
Mostly naming/nuance feedback, but it is worthwhile to consider keeping helm chart clean and straightforward for future updates

File: stable/application-mesh/templates/envoy-filter.yaml (hunk 1 above)
Comment (volkorny) [reply stable/application-mesh/templates/envoy-filter.yaml]: done, but we will have two filters in project like AMP: one for webserver and one for websocket

File: stable/application-mesh/templates/envoy-filter.yaml (hunk 2 above)
Comment (volkorny) [reply stable/application-mesh/templates/envoy-filter.yaml]: done

File: stable/application-mesh/templates/envoy-filter.yaml (hunk 3 above)
Comment (volkorny) [reply stable/application-mesh/templates/envoy-filter.yaml]: done

**Comment by volkorny** [comment]
app values example:
```
mesh:
  destinations:
    - component: webserver
      max_request_bytes: "67108864" # 64mb
    - component: websocket
      max_request_bytes: "67108864" # 64mb
```

File: stable/application-mesh/templates/envoy-filter.yaml [hunk 4]
```diff
@@ -1,36 +1,26 @@
...
-{{- if $ingress.settings.max_request_bytes }}
-{{- $host := required "Ingress host must be specified" $ingress.host }}
+{{- range $_, $destination := .Values.destinations }}
+{{- if $destination.max_request_bytes }}
 ---
 apiVersion: networking.istio.io/v1alpha3
 kind: EnvoyFilter
 metadata:
-  name: {{ include "application-helpers.name" $ }}-{{ include "application-helpers.uniq-id" $host }}
+  name: {{ include "application-helpers.name" $ }}-{{ $destination.component }}
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/templates/envoy-filter.yaml]: If we create a filter per destination - then the name should be something like `amp-webserver-request-size`. As we might add more filters in the future

File: stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml [hunk 5]
```diff
@@ -6,13 +6,12 @@ global:
 
 destinations:
   - component: webserver
+    max_request_bytes: "67108864"
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml]: let's nest this under settings same as it was before in ingerss
```
destinations:
   - component: webserver
     settings:
       max_request_bytes: "67108864"
```

**Review by volodymyr-mykhailyk — APPROVED** [review]
Just few minor comments again. Sorry for lot of back-and-forth

**Pull request merged**